### DISCUSSION TITLE
Embedded Bounty System/Points within RAWeb for players providing saves to assist with ticket fixes

### DISCUSSION THREAD

**Discussion by suspect15** [RetroAchievements/RAWeb#1579] [Ideas]
**Problem:** Many of the outstanding tickets are likely easy solves, but the amount of time needed for a developer to play through a game long enough to be able to reproduce the issue is a massive barrier.  Because of this, many tickets remain unresolved and the amount continues to accumulate throughout the year, typically peaking just prior to the start of Devember.  Events do not attract enough sustained interest/awareness to result in an overall trend of decreasing open ticket quantities.

The biggest issues to this problem is the lack of save file availability.  Understandably, many times players who create tickets don't have saves prior to when an achievement fails because they didn't anticipate a need for one.  By the time they realize the achievement is broken, they can't easily create a save needed to quickly reproduce an issue.

I think we are missing a clear opportunity to incentivize player assistance in a meaningful way by not considering what motivates RA users.

**Proposed Solution:** Incentivize unrestricted player assistance by incorporating a save file/state bounty system within RAWeb that meets the following requirements:

1. Communicates the need for saves on set pages by some obvious mechanism such as a subsection on the set page or perhaps a clearly visible link to an expanded section of the ticket system where developers can specify the parameters of the save they'd like to be provided.

2. Has some sort of repository available for players to submit saves.  Perhaps a new discord forum or direct upload link to RAWeb for small save files?  There may need to be different options for difference systems based on file sizes.

3. Awards "Bounty points" on player pages similar to Dev "Achievements won by others" stats.  At certain thresholds, tiered Site Award badges would be given to players who reach bounty point totals in a similar fashion to the Dev  achievement and point unlock badges.

4. Create a leaderboard on site to rank players by their accumulated bounty points.

**Bounty System Design Concept**

The idea behind the Bounty System is to allow users to be recognized for their direct contributions to resolving tickets and ultimately the increased quality and player experience of RA itself.  Once a save is submitted for a bounty, the ticket resolving developer must validate the submitted save was utilized in a meaningful way.  I think the best implementation method is expanding the ticket system interface to allow developers to specify exactly what type of saves they are requesting (the bounty), to track which user submits a save file (the claim) and a way for the resolving developer to validate the claim and award the bounty points.

I envision a player checking out a set page for a game they intend to play, immediately recognizing a bounty and deliberately making and submitting the needed saves during their playthrough.

As someone who tries to reduce open ticket volume, I don't see it ever getting any better than it currently is and the trend will not be favorable.  Over time the number of existing sets with inactive developers will inevitably increase.  The sooner we have a system that removes the single biggest barrier to resolving tickets, the better RA will be able to support high quality and continued growth.

**Comment by Jamiras** [comment]
> Has some sort of repository available for players to submit saves.

See also #712 and #1570

**Comment by suspect15** [comment]
As another suggestion - shy of a bounty system, a bulletin board of needed save files might be another way to crowdsource save file gathering assistance.

Suggestion:
- New subpage for each set that can list requested saves
- Way for developers to add needed saves to the list
- Way for requesting developers to be contacted by players with saves
- Prominent link to the subpage on set pages

With over 2,000 steady state tickets open at any given time, the need for something like this continues to increase.

**Comment by suspect15** [comment]
I took a poll in the RA discord to see how players might be willing to assist with submitting saves if they were aware of a need before beginning a set.  The results were overwhelmingly supportive with 62 of 65 respondents stating they would be willing.  https://discord.com/channels/310192285306454017/511718348178849792/1285328768709169223

![image](https://github.com/user-attachments/assets/f0807232-095b-4bc5-b44d-5b1dddc54c94)

**Comment by Tsearo** [comment]
Fully in support of this idea and think motivating players to help provide saves/states for tickets will be one the biggest steps in helping nail down the ticket backlog and keep inactive (and even active) developer sets in better shape. The only way I see RA continuing to be able to handle the enormous amount of tickets created is if the players are creating useful tickets including either good information and/or saves/states to replicate or at least examine a nearby area. The biggest blocker for developers as Suspect pointed out are usually time investment to play through the game or find a way to skip to later parts of a game so if players were providing this then a lot of tickets would be able to get reviewed in a much shorter time. Currently ticket events are typically the main time where we get players to help create saves for highly ticketed games but a more consistent and long-term solution would be to have permanent motivation for players to assist.

So Suspect's Bounty system seems like a perfect motivator which allows targeted focus on problematic sets in need of saves or states. This would work wonderfully as a feature maintained by QA or possibly the TicketHounds/TicketBusters groups. QA regularly monitors the most ticketed sets and when needed will reach out to developers to see if anyone is interested in tackling them but by being able to have players assist as well I think this would go even further to maintain these sets.

To add on to this a bit if there was a reasonable way to store and track submitted saves/states for achievements/games then I think it would be great if there was some indicator on a game page (and more importantly on any tickets for the game) if saves are needed or available. I don't have a good solution to how we could store these on the site directly as that raises a space concern (states for things like PSP are huge and not reasonable to upload), possibly urls but then you run into access issues and expired links. But if we could theoretically see on a ticket that a save is available (and even download it if dev privs) that would be amazing. I know some past ticket events have saved a repository of player saves but I would imagine most developers have no idea about this or which games have full-game saves submitted and available.

My main extra addition to this idea would be to also add a general user/player karma stat and tiered badge which increased based on valid ticket reports. This would work in addition/separately from the Bounty points suggested by Supsect as that could be treated more like a permanent event while general karma would be an overall statistic.

My additional proposal:
Add a simple rating system when a developer closes out a ticket where the developer can rate the usefulness of the ticket. Preferably keep it very simplified but also motivate players to provide good information/states.

Ex. (Rough starting point)
Please rate the quality of this ticket:
1* Terrible - Bad reports
2* Bad - Not reproducible and no helpful details provided by player
3* Okay - Found the issue but nothing extra provided by player other than pointing out it didn't work.
4* Great - User provided any of the following: helpful details, save/state, good communication
5* Amazing - User provided good information and saves/states

For 1-2 stars don't do anything other than track the rating
For 3 stars award 1 karma point
For 4 stars award 2 karma points
For 5 stars award 3 karma points

Alternatively it could have a checklist rating instead such as:
- [ ] Issue was valid (or at least feasible if not reproducible)
- [ ] User communicated in a helpful manner
- [ ] User provided valuable details which helped reproduce or resolve the issue
- [ ] User provided valuable saves or savestates to help troubleshoot the issue
Where each of these would award a point

Basically we don't want to demotivate players from creating tickets but we don't want to award anything for useless/bad tickets while motivating players to go beyond just submitting a ticket with no real information/save even if it is valid.

Depending on which data format is used when closing out the tickets we could even add a section to the monthly RANews posts which highlight top contributing players overall and per month by specific statistics (bounties, karma, saves submitted, valid tickets submitted, etc).

Overall RA definitely needs some form of incentive for players to help contribute to the growing amount of tickets we are seeing with the increased player-base.

**Comment by televandalist** [comment]
Should also consider and account for the various ways states will not work.

Some examples:

- Many states break between emulator/core updates, so it would be important to also have various versions of those available as well. Something like PCSX2 is easy since you can go back and download pretty much any stable/nightly build, but an older build of a libretro core isn’t as easy to track down. 

- BIOS set-ups can change things drastically. PC-8801 probably being the most notorious…

- Some states will not load if certain core settings do not match. First one that pops in my head is the Beetle Saturn core. A state will not work if your RAM expansion settings (aka the cart used) differ from the player who made the state. The best way to deal with this would probably be to require the core’s config file along with the state.
